ext/standard/mail.c

```diff
@@ -66,20 +66,32 @@ static bool php_mail_build_headers_check_field_value(zval *val)
 	/* https://tools.ietf.org/html/rfc2822#section-2.2.3 */
 	while (len < value->len) {
 		if (*(value->val+len) == '\r') {
+			if (*(value->val+len+1) != '\n') {
+				return CONTAINS_CR_ONLY;
+			}
+
 			if (value->len - len >= 3
-				&&  *(value->val+len+1) == '\n'
 				&& (*(value->val+len+2) == ' '  || *(value->val+len+2) == '\t')) {
 				len += 3;
 				continue;
 			}
-			return FAILURE;
+
+			return CONTAINS_CRLF;
+		}
+		if (*(value->val+len) == '\n') {
```

Comment:
This allows a header like `Subject: \n\tfoo`, but is that allowed by the RFC? From my understanding it is never allowed except when using folding, but for folding you have to use CRLF.
I could be wrong of course, I wanted to check this with you.

Reply:
@nielsdos 
Yes, you are right.

>From my understanding it is never allowed except when using folding, and for folding you have to use CRLF instead of a lone LF.

The reason for this pull request is that `LF` is often treated like `CRLF`. Therefore, the intention was to avoid breaking the environment of users who already use `LF` for folding.

It would be possible to strictly prohibit this, but IMHO I think it could lead to a BC Break.

Reply:
Okay I see, thanks for answering.